scipio/src/parking.rs

```diff
@@ -93,15 +95,23 @@ impl fmt::Debug for Parker {
     }
 }
 
-struct Inner {}
+struct Inner {
+    wakers: RefCell<Vec<Waker>>,
+}
 
 impl Inner {
-    fn park(&self, timeout: Option<Duration>) -> bool {
-        // If the timeout is zero, then there is no need to actually block.
-        // Process available I/O events.
-        let reactor_lock = Reactor::get().lock();
-        let _ = reactor_lock.react(timeout);
-        false
+    /// If the timeout is zero, then there is no need to actually block.
+    /// Process available I/O events.
+    fn poll(&self, timeout: Option<Duration>) {
+        let _ = Reactor::get()
+            .lock()
```

Comment:
if you are down for a cleanup, the lock stuff in the reactor can go.
It is an artifact of async-io as well.
It may simplify your code: you may not need the refcell on the vector if you dispose of it.